### DISCUSSION TITLE
Is it possible to filter S3 objects that are publicly accessible? In a non public S3 resource

### DISCUSSION THREAD

**Discussion by Ghosted-droid** [cloud-custodian/cloud-custodian#7360] [General]
I want to check if at least one of objects in an S3 bucket is publicly accessible.

For example:
 where S3 bucket is non-public (in Bucket ACL and Account ACL) but one of it's objects is publicly accessible.

Thank you for your time

**Comment by kapilt** [comment]
> I want to check if at least one of objects in an S3 bucket is publicly accessible.
> 
> For example:
> where S3 bucket is non-public (in Bucket ACL and Account ACL) but one of it's objects is publicly accessible.
> 
> Thank you for your time

you can use tools/c7n_salactus for scanning individual objects at scale for acl, encryption, etc, it was originally a capability within custodian but for any large object count (billions, many buckets, many accounts), it made more sense to split out as a separate tool.

see https://github.com/cloud-custodian/cloud-custodian/tree/master/tools/c7n_salactus

**Comment by kapilt** [comment]
also using s3 public block can cause acls to be ignored on individual objects
